SmartDeviceLink/SDLMenuManager.m

```diff
@@ -0,0 +1,415 @@
+    [cells enumerateObjectsUsingBlock:^(SDLMenuCell * _Nonnull cell, NSUInteger index, BOOL * _Nonnull stop) {
+        if (cell.subCells.count > 0) {
+            [mutableCommands addObject:[self sdl_subMenuCommandForMenuCell:cell position:(UInt16)index]];
+        } else {
+            [mutableCommands addObject:[self sdl_commandForMenuCell:cell withArtwork:shouldHaveArtwork position:(UInt16)index]];
+        }
+    }];
+
+    return [mutableCommands copy];
+}
+
+- (NSArray<SDLRPCRequest *> *)sdl_subMenuCommandsForCells:(NSArray<SDLMenuCell *> *)cells withArtwork:(BOOL)shouldHaveArtwork {
+    NSMutableArray<SDLRPCRequest *> *mutableCommands = [NSMutableArray array];
+    for (SDLMenuCell *cell in cells) {
+        if (cell.subCells.count > 0) {
+            [mutableCommands addObjectsFromArray:[self sdl_allCommandsForCells:cell.subCells withArtwork:shouldHaveArtwork]];
+        }
+    }
+
+    return [mutableCommands copy];
+}
+
+- (NSArray<SDLRPCRequest *> *)sdl_allCommandsForCells:(NSArray<SDLMenuCell *> *)cells withArtwork:(BOOL)shouldHaveArtwork {
+    NSMutableArray<SDLRPCRequest *> *mutableCommands = [NSMutableArray array];
+    [cells enumerateObjectsUsingBlock:^(SDLMenuCell * _Nonnull cell, NSUInteger index, BOOL * _Nonnull stop) {
+        if (cell.subCells.count > 0) {
+            [mutableCommands addObject:[self sdl_subMenuCommandForMenuCell:cell position:(UInt16)index]];
+            [mutableCommands addObjectsFromArray:[self sdl_allCommandsForCells:cell.subCells withArtwork:shouldHaveArtwork]];
+        } else {
+            [mutableCommands addObject:[self sdl_commandForMenuCell:cell withArtwork:shouldHaveArtwork position:(UInt16)index]];
+        }
+    }];
+
+    return [mutableCommands copy];
+}
+
+- (SDLAddCommand *)sdl_commandForMenuCell:(SDLMenuCell *)cell withArtwork:(BOOL)shouldHaveArtwork position:(UInt16)position {
+    SDLAddCommand *command = [[SDLAddCommand alloc] init];
+
+    SDLMenuParams *params = [[SDLMenuParams alloc] init];
+    params.menuName = cell.title;
+    params.parentID = cell.parentCellId != UINT32_MAX ? @(cell.parentCellId) : nil;
+    params.position = @(position);
+
+    command.menuParams = params;
+    command.vrCommands = cell.voiceCommands;
+    command.cmdIcon = (cell.icon && shouldHaveArtwork) ? [[SDLImage alloc] initWithName:cell.icon.name] : nil;
+    command.cmdID = @(cell.cellId);
+
+    return command;
+}
+
+- (SDLAddSubMenu *)sdl_subMenuCommandForMenuCell:(SDLMenuCell *)cell position:(UInt16)position {
+    SDLAddSubMenu *submenu = [[SDLAddSubMenu alloc] initWithId:cell.cellId menuName:cell.title];
+    submenu.position = @(position);
+
+    return submenu;
+}
+
+#pragma mark - Observers
+
+- (void)sdl_commandNotification:(SDLRPCNotificationNotification *)notification {
+    SDLOnCommand *onCommand = (SDLOnCommand *)notification.notification;
+
+    for (SDLMenuCell *cell in self.menuCells) {
+        if (onCommand.cmdID.unsignedIntegerValue != cell.cellId) { continue; }
```

Comment:
The handlers for submenu `SDLMenuCell`s are not getting called because the submenu cells are not being searched.